### ISSUE TITLE
Run Keyword And Warn On Failure keyword

### ISSUE THREAD

**Issue by dimanhsb** [robotframework/robotframework#2294]
Hello,

A useful feature (that we have already implemented in our framework, but might be useful for everyone) is a "Run Keyword And Warn On Failure" keyword - same thing as Run Keyword And Continue On Failure, but KW passes, and issues a warning.

our current implementation in Robot Framework is:

```
Run Keyword And Warn On Failure
    [Arguments]  ${kw}  @{args}
    [Documentation]    When keyword fails, this passes, but a warning is issued.
    ${state}  ${output}=  Run Keyword And Ignore Error  ${kw}  @{args}
    Run Keyword If  '${state}'!='PASS'  Log  Warning: Keyword failed: ${kw} @{args}{\n}${output}  WARN
    [Return]  ${output}
```

**Comment by pekkaklarck** [comment]
This should work like `Run Keyword And Ignore Error` but also report possible failure as a warning, right? I guess it would be useful where you absolutely don't want the keyword to fail but don't want it fail silently either.

I want to be fairly conservative when accepting new BuiltIn keywords, but perhaps this would be useful enough. An alternative is making it somehow possible to configure the existing `Run Keyword And Ignore Error` to support this. What do others think? If we agree this keyword should be added, would someone be interested to provide a pull request implementing it?

**Comment by HelioGuilherme66** [comment]
I think the best keyword to add the warning option is the `Run Keyword And Continue On Failure` not the `Run Keyword And Ignore Error`.

**Comment by pekkaklarck** [comment]
I don't see why this would be so useful with `Run Keyword And Continue On Failure`. It doesn't hide the failure. It just allows execution to continue afterwards. If we decide to support this functionality by configuring existing keywords, adding such support also to that keyword wouldn't hurt. Adding a new keyword, like originally proposed, may be easier and clearer, though.

**Comment by dimanhsb** [comment]
Yes, I agree that there might be some confusion on a keyword that would be `Run Keyword And Ignore Error But Warn` or the like.

If it is to be added, I think it's quite clear what  `Run Keyword And Warn On Failure` means, so it should probably be added as-is.

**Comment by dimanhsb** [comment]
I have implemented the keyword in python, but I haven't been able to understand the testing framework on how to create relevant unit tests (other than a simple one issuing a failure and expecting the test case to pass). Should I do the pull request, and wait for others to fill in?

**Comment by pekkaklarck** [comment]
Did you read atest/README.rst?

**Comment by dimanhsb** [comment]
Yes I did, unfortunately I don't have the experience required to follow the acceptance framework.

**Comment by pekkaklarck** [comment]
Any new info about this? I'm fine adding the requested keyword but we need a pull request. See [contribution guidelines](https://github.com/robotframework/robotframework/blob/master/CONTRIBUTING.rst) for more details.

**Comment by pekkaklarck** [comment]
One use case for this keyword would be turning failing suite teardowns to warnings to avoid all tests being marked failed. A benefit of that instead of using `Run Keyword And Ignore Error` is that you'd still see the warning in the log file. If someone would be interested to implement this, this could be added already in RF 3.1.

As a somewhat related enhancement, we could also add some kind of a note in the report that there are failures in the log file. Currently you can't see that at all in report, so warnings in logs may be accidentally ignored. This obviously requires a separate issue.

**Comment by pekkaklarck** [comment]
This was implemented in PR #3670 by @cristii006. Thanks for contribution!

**Comment by terokinnunen** [comment]
> This was implemented in PR #3670 by @cristii006. Thanks for contribution!

YM #3679.

**Comment by pekkaklarck** [comment]
True, this was implemented by #3679.
